be/src/exprs/time_functions.cpp

```diff
@@ -2378,6 +2378,32 @@ StatusOr<ColumnPtr> TimeFunctions::make_date(FunctionContext* context, const Col
     return result.build(ColumnHelper::is_all_const(columns));
 }
 
+// date_diff
+StatusOr<ColumnPtr> TimeFunctions::datediff(FunctionContext* context, const Columns& columns) {
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+    ColumnViewer<TYPE_DATETIME> lv_column(columns[0]);
+    ColumnViewer<TYPE_DATETIME> rv_column(columns[1]);
+    ColumnViewer<TYPE_VARCHAR> type_column(columns[2]);
```

Comment:
if type_column is const column, we should handle it specialy to avoid duplicate computation, you can refer to the implementation of sha2 function to optimize it.

Reply:
ok，get it